### PR TITLE
support for advanced styling

### PR DESCRIPTION
**Pull request by adventmedia**
based on current usage for the Radio button component, which adds/removes a class to a wrapper div based on the selected state of the button; applied to Checkbox component as well.
Default styling is not affected.

### CODE REVIEW

**Comment by adventmedia** [comment]
@travist wondering what sort of timeline we can hope for this to be merged and released?

File: src/components/checkbox/Checkbox.js [hunk 1]
```diff
@@ -177,6 +177,18 @@ export default class CheckBoxComponent extends Field {
     return false;
   }
 
+  updateValue(value, flags) {
+    const checkBoxChecked = 'checkbox-checked';
+    const val = this.getValue();
+    if (val) {
+      this.addClass(this.refs.formCheck, checkBoxChecked);
```
Comment (travist) [inline src/components/checkbox/Checkbox.js:184]: Need to protect against templates that do not have this reference. It will crash otherwise.

File: src/components/radio/Radio.js [hunk 2]
```diff
@@ -158,12 +158,26 @@ export default class RadioComponent extends Field {
...
+          case 'radio' :
         if (input && input.value.toString() === value.toString()) {
           //add class to container when selected
           this.addClass(wrapper, optionSelectedClass);
         }
         else {
           this.removeClass(wrapper, optionSelectedClass);
+        }
+            break;
+          case 'checkbox' :
```
Comment (travist) [inline src/components/radio/Radio.js:171]: When would a "radio" ever be a checkbox?  I believe this should be using the Select Boxes component instead.

File: src/components/radio/Radio.js [hunk 3]
```diff
@@ -178,7 +192,7 @@ export default class RadioComponent extends Field {
       && this.previousValue === this.currentValue;
     if (shouldResetValue) {
       this.resetValue();
-      this.triggerChange(flags);
+      this.triggerChange();
```
Comment (travist) [inline src/components/radio/Radio.js:195]: Why was this changed? This will certainly introduce a bug somewhere if we accept this. We should be passing along the flags since they are commonly used to flag behaviors during the onChange event loop.

File: src/templates/bootstrap/radio/form.ejs [hunk 4]
```diff
@@ -1,6 +1,8 @@
 <div class="form-radio radio">
   {% ctx.values.forEach(function(item) { %}
-  <div class="form-check{{ctx.inline ? '-inline' : ''}}" ref="wrapper">
+  <div class="form-check{{ctx.inline ? '-inline' : ''}} {% if ((ctx.disabled && ctx.value) && (ctx.value === item.value || (typeof ctx.value === 'object' && ctx.value.hasOwnProperty(item.value) && ctx.value[item.value]))) { %}
```
Comment (travist) [inline src/templates/bootstrap/radio/form.ejs:3]: This much logic should be placed within the class and then passed into the template as a ctx variable.

**Review by travist — CHANGES_REQUESTED** [review]
I feel a little weary about pulling this in. I gave some general comments so please respond to them so that I can understand your motivation for this change.

File: src/components/radio/Radio.js (hunk 2 above)
Comment (adventmedia) [reply src/components/radio/Radio.js:193]: this is to support SelectBoxes component
`export default class SelectBoxesComponent extends RadioComponent`

File: src/components/radio/Radio.js (hunk 3 above)
Comment (adventmedia) [reply src/components/radio/Radio.js]: fixed (working from 4.12.4)

File: src/templates/bootstrap/radio/form.ejs (hunk 4 above)
Comment (adventmedia) [reply src/templates/bootstrap/radio/form.ejs]: this is done

**Comment by adventmedia** [comment]
these changes are to support CSS restyling of the input components (esp. for mobile devices). It is necessary to add or remove a class from the "wrapper" div based on whether the input is selected or not. Radio component already supports this behaviour.
screenshot_134
